policy-engine/src/openapi.rs

```diff
@@ -0,0 +1,22 @@
+use actix_web::{HttpRequest, HttpResponse};
+use string_template::Template;
+use AppState;
+
+pub(crate) fn index(req: HttpRequest<AppState>) -> HttpResponse {
+    let path_prefix = &req.state().path_prefix;
+    let spec_template = include_str!("openapiv3.json");
+    let spec = Template::new(spec_template).render_named(
+        &[("path_prefix", path_prefix.as_str())]
+            .iter()
+            .cloned()
+            .collect::<std::collections::HashMap<&str, &str>>(),
+    );
```

Comment:
All of this seems a bit overkill to me for a single parameter fill in. Why not just a `format!()` which also does the check statically? (may need a wrapping macro if the template is not a literal)

Reply:
With `format!()` we'd have to escape every curly brace with another curly brace. I prefer to have the spec template as close to the original syntax as possible which is achieved with the current method. When you say overkill, which metric are you talking about? Since the development time is already spent anyway, do you see performance drawbacks compared to another method?

Reply:
Logical complexity. As in, I would have expected the simpler, static approach with no external dependencies first. But your explanation makes sense, I'm ok with sticking to it.

This isn't a core hotpath and should be easily cacheable (both on our and on proxy side), so I'm not too concerned with performances here.

Reply:
Another approach which I'm considering is to omit the template strings in the specification and use an OpenAPI parser which understands the spec and can rewrite the paths before serializing it again.

Reply:
That implementation looks nice. Comments I have on that are:
 * it goes through a deserialization+serialization roundtrip on each request, but as above I don't think performance is a concern here
 * runtime result depends on correctness of the library. I am not familiar with it, I hope you did a manual sanity check on the result